src/main/scala/chisel3/util/BitSet.scala

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+trait BitSet { bsf =>
```

Comment:
> Why using bsf instead of more canonical self or just built-in this?

Ah I realize now it has to do with the anonymous inner classes and needing to distinguish. Nevertheless I think `self` is more canonical or `outer` is perhaps more precise.

Reply:
I switch to `outer`